docs/security.rst

```diff
@@ -310,3 +310,20 @@ standard port 443, you'll need to configure that too. Be aware that super user p
     # Optionally, set the server to listen on the standard SSL port.
     web_server_port = 443
     base_url = http://<hostname or IP>:443
+
+Impersonation
+'''''''''''''
+
+Airflow has the ability to impersonate a unix user while running task
+instances based on the task's ``run_as_user`` parameter, which takes a user's name.
+
+*NOTE* For impersonations to work, Airflow must be run with `sudo` as subtasks are run
```

Comment:
(Keeping my comment here for to make sure it stays in thread until a summary is available).

My idea around passing the configuration (in the future?) is 1) let the task pick up its config through the Rest API. This allows for better security and doesn't need the config files and accompanying operational issues (deletion, creation, accessibility etc), 2) let it be passed by the scheduler to the executor to the task. This removes a reliance on having an API available somewhere and makes task instantiation atomic (ie. everything is available at the moment the task launches) or 3) use a kind of RPC call over the messaging bus. Which is kind of the same as 1 but might be easier to scale.

On the issue of airflow.cfg needing root access, I am not sure I understand why yet and I would really like to work with you to remove this requirement. To help this further I can for example write up an example sudoers file that should work out of the box.